plugins/inputs/nats_consumer/nats_consumer.go

```diff
@@ -112,19 +122,25 @@ func (n *natsConsumer) Start(acc telegraf.Accumulator) error {
 		n.errs = make(chan error)
 		n.Conn.SetErrorHandler(n.natsErrHandler)
 
-		n.in = make(chan *nats.Msg)
+		n.in = make(chan *nats.Msg, 1000)
 		for _, subj := range n.Subjects {
-			sub, err := n.Conn.ChanQueueSubscribe(subj, n.QueueGroup, n.in)
+			sub, err := n.Conn.QueueSubscribe(subj, n.QueueGroup, func(m *nats.Msg) {
+				n.in <- m
+			})
```

Comment:
Think it would help to add a `n.Conn.Flush()` here too just in case to ensure that the subscription has been processed by the server.